packages/core/src/components/button/abstractButton.tsx

```diff
@@ -39,7 +39,7 @@ export interface IButtonProps extends IActionProps {
     alignText?: Alignment;
 
     /** A ref handler that receives the native HTML element backing this component. */
-    elementRef?: (ref: HTMLElement | null) => any;
+    elementRef?: React.Ref<HTMLElement>;
```

Comment:
I don't love this approach because this type reference will not compile with `@types/react@^15` (the latest version of which was published in October 2019).

There may be some React 15 users of Blueprint who use those typings (albeit I have not tested this myself, whether Blueprint compiles with `@types/react@^15`, but judging by a quick search of our references to react types across the monorepo, I think it does)

You should try to make this change without referencing the new `React.Ref` type, maybe with a new function in a utils file which helps you get the ref across React versions:

```ts
// interface which is assignable to React.Ref<T>, but does not require us to depend on @types/react@^16
export interface IRefObject<T> {
    readonly current: T | null;
}

export function getRef<T>(ref: any) {
    return isFunction(ref) ? ref as (r: T | null) => any
        : typeof ref.current !== "undefined" ? ref as IRefObject<T>
        : null;
}
```

Reply:
Done